spec/domain/jubla/person/event_queries_spec.rb

```diff
@@ -0,0 +1,46 @@
+# encoding: utf-8
+
+#  Copyright (c) 2012-2015, Pfadibewegung Schweiz. This file is part of
+#  hitobito and licensed under the Affero General Public License version 3
+#  or later. See the COPYING file at the top-level directory or at
+#  https://github.com/hitobito/hitobito.
+
+require 'spec_helper'
+
+describe Person::EventQueries do
+  let(:person) { participation.person }
+  let(:event) { participation.event }
+  let(:participation) { role.participation }
+
+  context :coach do
+    let(:role) { Fabricate(:event_role, type: 'Event::Camp::Role::Coach' )}
+
+    it 'can see event 20 days ago' do
+      event.dates.first.update(finish_at: Time.now - 20.day)
+      p = PersonDecorator.new(person)
+      expect(p.coached_events.count).to eq(1)
+    end
+
+    it 'can not see event 40 days ago' do
+      event.dates.first.update(finish_at: Time.now - 40.day)
+      p = PersonDecorator.new(person)
+      expect(p.coached_events.count).to eq(0)
+    end
+  end
+
+  context :assistant_leader do
+    let(:role) { Fabricate(:event_role, type: 'Event::Role::AssistantLeader' )}
+
+    it 'can see event 20 days ago' do
+      event.dates.first.update(finish_at: Time.now - 20.day)
+      p = PersonDecorator.new(person)
+      expect(p.coached_events.count).to eq(1)
+    end
+
+    it 'can not see event 40 days ago' do
+      event.dates.first.update(finish_at: Time.now - 40.day)
+      p = PersonDecorator.new(person)
+      expect(p.coached_events.count).to eq(0)
+    end
+  end
+end
```

Comment:
Ich meinte, die Tests für die Zeitspannen in ein `event_spec.rb` auszulagern, wo die Methode `up_to_a_month_ago` direkt getestet wird. Das `event_queries_spec.rb` beibehalten und Tests schreiben, welche die verschiedenen Aspekte der `coached_events` Methode überprüfen (verschiedene Rollen, Sortierung, Zeitspanne, ...)